app/main.py

```diff
@@ -2,0 +1,26 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name, health=100, hidden=False):
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def die(self):
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self):
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+class Herbivore(Animal):
+    def hide(self):
+        self.hidden = not self.hidden
```

Comment:
The `hide` method toggles the `hidden` state of the `Herbivore`. This is correct, but ensure that the logic for hiding is consistent with the intended behavior of the application.